docs/docs/connectors/custom-connectors.mdx

```diff
@@ -5,70 +5,199 @@ title: Custom Connectors
 description: Deploy and Run a Rasa Chat Bot on a custom chat interface
 ---
 
-You can also implement your own custom channel. You can
-use the `rasa.core.channels.rest.RestInput` class as a template.
-The methods you need to implement are `blueprint` and `name`. The method
-needs to create a sanic blueprint that can be attached to a sanic server.
 
-This allows you to add REST endpoints to the server that the external
-messaging service can call to deliver messages.
+You can implement your own custom channel connector as a python class. You can
+use the `rasa.core.channels.rest.RestInput` class as a template. 
 
-Your blueprint should have at least the two routes: `health` on `/`,
-and `receive` on the HTTP route `/webhook`.
+A custom connector class must subclass `rasa.core.channels.channel.InputChannel`
+and implement at least a `blueprint` and `name` method. 
 
-The `name` method defines the url prefix. For example, if your component is
-named `myio`, the webhook you can use to attach the external service is
+
+## The `name` method
+
+The `name` method defines the url prefix for the connector's webhook. It also defines the channel name you should use 
+in any [channel specific response variations](../responses.mdx#channel-specific-response-variations) and the name you
+should pass to the `output_channel` query parameter on the [trigger intent endpoint](https://www.rasa.com/docs/rasa/pages/http-api#operation/triggerConversationIntent)
+
+For example, if your custom channel is named `myio`, you would define the `name` method as:
+
+```python title=custom_channel.py
+from rasa.core.channels.channel import InputChannel
+
+class MyIO(InputChannel):
+    def name() -> Text:
+        """Name of your custom channel."""
+        return "myio"
+```
+
+You would write a response variation specific to the `myio` channel  as:
+
+```yaml-rasa title="domain.yml"
+responses:
+  utter_greet:
+    - text: Hi! I'm the default greeting.
+    - text: Hi! I'm the  custom channel greeting
+      channel: myio
+```
+
+The webhook you give to the custom channel to call would be
 `http://<host>:<port>/webhooks/myio/webhook`, replacing 
 the host and port with the appropriate values from your running Rasa X or Rasa Open Source server.
 
-You can `POST` messages to your custom connector at the url 
-`http://<host>:<port>/webhooks/myio/webhook` using the following format:
 
-```json
-{
-  "sender": "test_user",  // sender ID of the user sending the message
-  "message": "Hi there!",
-  "metadata": {}  // optional, any extra info you want to add for processing in NLU or custom actions
-}
+## The `blueprint` method
+
+The `blueprint` method
+needs to create a [sanic blueprint](https://sanic.readthedocs.io/en/v20.3.0/sanic/blueprints.html#blueprints)
+that can be attached to a sanic server. 
+Your blueprint should have at least the two routes: `health` on the route `/`,
+and `receive` on the route `/webhook`.
```

Comment:
I'd add here that there's an example further down, as this isn't immediately clear and the example requires a bit of scrolling to get to